extensions/machinepools/machinepools.go

```diff
@@ -301,11 +302,23 @@ func MatchRoleToPool(poolRole string, allRoles []Roles) int {
 
 // GetInitMachine accepts a client and clusterID and returns the "init node" machine
 // object for rke2/k3s clusters
-func GetInitMachine(client *rancher.Client, clusterID string) (*v1.SteveAPIObject, error) {
+func GetInitMachine(client *rancher.Client, clusterType, clusterID string) (*v1.SteveAPIObject, error) {
+	var secret *secrets.SecretList
+	var err error
+
 	logrus.Info("Retrieving secret and identifying machine...")
-	secret, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{
-		LabelSelector: initNodeLabelKey + "=" + True,
-	})
+
+	if strings.Contains(clusterType, "RKE2") {
+		clusterID = strings.Replace(clusterID, "fleet-default/", "", 1)
+
+		secret, err = secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{
+			LabelSelector: initNodeLabelKey + "=" + True + "," + clusterNameLabel + "=" + clusterID,
```

Comment:
I'm sure this was a result of your testing but, is there a reason just `initNodeLabelKey` isn't selecting the right node on rke2? Wouldn't you want the clusterID selector added in both rke2 and k3s?

Reply:
why did it panic?

Reply:
resolved